apps/cowswap-frontend/src/modules/hooksStore/pure/AddCustomHookForm/index.tsx

```diff
@@ -20,52 +20,113 @@ interface AddCustomHookFormProps {
   const [isSearchOpen, setSearchOpen] = useState<boolean>(false)
   const [isWarningAccepted, setWarningAccepted] = useState<boolean>(false)
-  const [isUrlValid, setUrlValid] = useState<boolean>(true)
   const [loading, setLoading] = useState<boolean>(false)
   const [isFinalStep, setFinalStep] = useState<boolean>(false)
   const [manifestError, setManifestError] = useState<string | React.ReactNode | null>(null)
   const [dappInfo, setDappInfo] = useState<HookDappIframe | null>(null)
 
-  // Function to reset all states
   const dismiss = useCallback(() => {
     setDappInfo(null)
-    setManifestError(null)
-    setUrlValid(true)
     setLoading(false)
     setFinalStep(false)
     setWarningAccepted(false)
   }, [])
 
-  // Function to handle going back from search mode
   const goBack = useCallback(() => {
     dismiss()
-    setInput(undefined)
+    setManifestError(null)
+    setInput('')
     setSearchOpen(false)
   }, [dismiss])
 
-  // Function to handle adding the hook dapp
   const addHookDappCallback = useCallback(() => {
     if (!dappInfo) return
     addHookDapp(dappInfo)
     goBack()
   }, [addHookDapp, dappInfo, goBack])
 
-  // Effect to validate the URL whenever the input changes
-  useEffect(() => {
-    // Reset state whenever input changes
-    dismiss()
-    setUrlValid(input ? uriToHttp(input).length > 0 : true)
-  }, [input, dismiss])
+  // Normalizes URLs only on explicit actions (paste/submit) to prevent interrupting user typing
+  const normalizeUrl = useCallback((url: string, shouldNormalize = false) => {
+    if (!url) return ''
+
+    if (shouldNormalize) {
+      try {
+        const normalizedUrl = url.replace(/\/+$/, '')
+
+        if (normalizedUrl.startsWith('https://')) {
+          return normalizedUrl
+        }
+
+        if (normalizedUrl.startsWith('http://')) {
+          return 'https://' + normalizedUrl.slice(7)
+        }
+
+        // Handle special protocols (ipfs/ipns/ar) via uriToHttp
+        const urls = uriToHttp(normalizedUrl)
+        if (urls.length > 0) {
+          return urls[0]
+        }
+
+        return normalizedUrl
+      } catch (error) {
+        console.warn('Invalid URL during normalization:', error)
+        return url
+      }
+    }
+
+    return url
+  }, [])
+
+  const handleInputChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value
+    // Clear error when input changes
+    setManifestError(null)
+    setInput(value)
+    setDappInfo(null)
+    setLoading(false)
+    setFinalStep(false)
+    setWarningAccepted(false)
+  }, [])
+
+  const handlePaste = useCallback(
+    (e: React.ClipboardEvent<HTMLInputElement>) => {
+      const pastedValue = e.clipboardData.getData('text')
+      const normalizedValue = normalizeUrl(pastedValue, true)
+      e.preventDefault() // Prevent default to avoid double paste
+      setInput(normalizedValue)
+      // Clear error on paste
+      setManifestError(null)
```

Comment:
Those calls are the same in handleInputChange(), it makes sens to move them in some callback:

```
const resetStates = useCallback(() => {
    setManifestError(null)
    setInput(value)
    setDappInfo(null)
    setLoading(false)
    setFinalStep(false)
    setWarningAccepted(false)
}, [])
```

Reply:
Addressed.